test/micro/org/openjdk/bench/jdk/incubator/vector/bigdata/SelectiveStore.java

```diff
@@ -0,0 +1,189 @@
+import java.util.concurrent.ThreadLocalRandom;
+
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorOperators;
+
+import org.openjdk.jmh.annotations.*;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 10, time = 1)
+@Fork(value = 1, jvmArgsPrepend = {"--add-modules=jdk.incubator.vector", "--add-exports", "java.base/jdk.internal.vm.vector=ALL-UNNAMED", "-ea"})
+public class SelectiveStore {
+
+  private static int ARRAY_LENGTH = 1024;
+
+  private static final VectorSpecies<Integer> INT_128_SPECIES =
+    IntVector.SPECIES_128;
+  private static final VectorSpecies<Integer> INT_256_SPECIES =
+    IntVector.SPECIES_256;
+  private static final VectorSpecies<Integer> INT_512_SPECIES =
+    IntVector.SPECIES_512;
+  private static final VectorSpecies<Integer> INT_PREFERRED_SPECIES =
+    IntVector.SPECIES_PREFERRED;
+
+  @State(Scope.Thread)
+  public static class ThreadLocalData {
+
+    private int conflict_scalar_cnt = 0;
+    private int conflict_vector_cnt = 0;
+    private int[] index = new int[ARRAY_LENGTH];
+    private int[] conflict_scalar = new int[ARRAY_LENGTH];
+    private int[] conflict_vector = new int[ARRAY_LENGTH];
+    private int[] input1 = new int[ARRAY_LENGTH];
+    private int[] input2 = new int[ARRAY_LENGTH];
+    private Random rand = new Random();
+
+    @Setup(Level.Trial)
+    public void doSetup() {
+      int conflict_id_each_5_elements = ThreadLocalRandom.current().nextInt(5);
+
+      for (int i = 0; i < ARRAY_LENGTH; i++) {
+        index[i] = ThreadLocalRandom.current().nextInt(ARRAY_LENGTH);
+        input1[i] = ThreadLocalRandom.current().nextInt();
+        // Generate 20% conflict data
+        int remainder = i % 5;
+        if (remainder == conflict_id_each_5_elements) {
+          input2[i] = ThreadLocalRandom.current().nextInt();
+        } else {
+          input2[i] = input1[i];
+        }
+
+        conflict_scalar[i] = 0;
+        conflict_vector[i] = 0;
+      }
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_scalar(ThreadLocalData tld) {
+    tld.conflict_scalar_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i++) {
+      if (tld.input1[i] != tld.input2[i]) {
+        tld.conflict_scalar[tld.conflict_scalar_cnt++] = tld.index[i];
+      }
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_vector_128(ThreadLocalData tld) {
+    tld.conflict_vector_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i += INT_128_SPECIES.length()) {
+      IntVector av = IntVector.fromArray(INT_128_SPECIES, tld.input1, i);
+      IntVector bv = IntVector.fromArray(INT_128_SPECIES, tld.input2, i);
+      IntVector cv = IntVector.fromArray(INT_128_SPECIES, tld.index, i);
+      VectorMask<Integer> mask = av.compare(VectorOperators.NE, bv);
```

Comment:
Can we mover vector compare out of benchmarking loop to reduce the noise.

Reply:
Thanks Jatin for your review.
This test I added is placed together with other benchmarks extracted from user cases.
The mask computation inside the loop is the essential logic in real sceanario.
We can add micro benchmarks in test/micro/org/openjdk/bench/jdk/incubator/vector/operation and then evaluate performance for compression / expanding in single API level.

Reply:
Ok got it, thanks! 
BTW can you also post the AVX2 numbers just to see how does it compare against scalar version when direct compress/expand instructions are not supported by target.

Reply:
FYI
![image](https://user-images.githubusercontent.com/70769035/136771782-cfb8adcf-7336-4173-a617-e2b89eadb1cb.png)